### DISCUSSION TITLE
Changelogs

### DISCUSSION THREAD

**Discussion by coot** [input-output-hk/cardano-engineering-handbook#30] [Ideas]
The consensus & network teams maintain [interface-CHANGELOG](https://github.com/input-output-hk/ouroboros-network/blob/master/ouroboros-consensus/docs/interface-CHANGELOG.md) (and [here](https://github.com/input-output-hk/ouroboros-network/blob/master/docs/interface-CHANGELOG.md)). These are tailored for downstream components, which allows us to skip changes which are not relevant for them.

For packages that are standalone libraries and used more widely this is not a good choice (and maybe it's not even good choice for the network & consensus anymore), e.g. things like `cardano-prelude` or packages which will land on Hackage.

**Comment by michaelpj** [comment]
The handbook already recommends that you should have a CHANGELOG or one per package: https://input-output-hk.github.io/cardano-engineering-handbook/policy/project/index.html#recommended-documents

It doesn't say anything about what goes in there, I'm somewhat happy to leave that to individual projects to decide, what do you think?

**Reply by coot** [reply]
I would put extra requirements for projects that are published on Hackage, for internal projects I think it's indeed better to leave things fluid.

**Reply by michaelpj** [reply]
https://github.com/input-output-hk/cardano-engineering-handbook/pull/33

**Comment by michaelpj** [comment]
That said, we could totally have a "Practices" section on CHANGELOGS, especially if we have some good examples.
